subprojects/packagefiles/openssl/meson.build

```diff
@@ -0,0 +1,406 @@
+    arch_subdir = 'linux-armv4'
+  elif host_machine.cpu_family() == 'aarch64' and host_machine.system() in ['linux', 'android']
+    arch_subdir = 'linux-aarch64'
+  elif host_machine.cpu_family() == 'x86' and host_machine.system() in ['dragonfly', 'freebsd', 'netbsd', 'openbsd']
+    arch_subdir = 'BSD-x86'
+  elif host_machine.cpu_family() == 'x86' and host_machine.system() in ['linux', 'android']
+    arch_subdir = 'linux-elf'
+  elif host_machine.cpu_family() == 'x86' and host_machine.system() == 'darwin'
+    arch_subdir = 'darwin-i386-cc'
+  elif host_machine.cpu_family() == 'x86' and host_machine.system() == 'sunos'
+    arch_subdir = 'solaris-x86-gcc'
+  elif host_machine.cpu_family() == 'x86' and host_machine.system() == 'windows'
+    arch_subdir = 'VC-WIN32'
+  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() in ['dragonfly', 'freebsd', 'netbsd', 'openbsd']
+    arch_subdir = 'BSD-x86_64'
+  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() == 'darwin'
+    arch_subdir = 'darwin64-x86_64-cc'
+  elif host_machine.cpu_family() == 'aarch64' and host_machine.system() == 'darwin'
+    arch_subdir = 'darwin64-arm64-cc'
+  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() == 'sunos'
+    arch_subdir = 'solaris64-x86_64-gcc'
+  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() == 'windows'
+    arch_subdir = 'VC-WIN64A'
+  elif host_machine.cpu_family() == 'aarch64' and host_machine.system() == 'windows'
+    arch_subdir = 'VC-WIN64-ARM'
+  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() == 'linux'
+    arch_subdir = 'linux-x86_64'
+  elif host_machine.cpu_family() == 'mips64' and host_machine.system() == 'linux'
+    arch_subdir = 'linux64-mips64'
+  else
+    error('Unsupported arch+OS combo: ' + host_machine.cpu_family() + ' + ' + host_machine.system())
+  endif
+
+  message('OpenSSL is configured without ASM support')
+else
+  message('OpenSSL is configured with ASM support')
+endif
+
+subdir('generated-config/archs' / arch_subdir / asm)
+
+# Build options specific to OS
+if host_machine.system() == 'windows'
+  defines += [
+    ## default of Win. See INSTALL in openssl repo.
+    'OPENSSLDIR="C:\\Program Files\\Common Files\\SSL"',
+    'ENGINESDIR="NUL"',
+    'OPENSSL_SYS_WIN32', 'WIN32_LEAN_AND_MEAN', 'L_ENDIAN',
+    '_CRT_SECURE_NO_DEPRECATE', 'UNICODE', '_UNICODE',
+  ]
+  if compiler.get_id() == 'msvc'
+    c_args += [
+      '-wd4090', '-Gs0', '-GF', '-Gy', '-nologo',
+    ]
+  endif
+elif host_machine.system() == 'darwin'
+  defines += [
+    'OPENSSLDIR="/System/Library/OpenSSL/"',
+    'ENGINESDIR="/dev/null"',
+  ]
+  c_args += [
+    '-Wno-missing-field-initializers',
+  ]
+elif host_machine.system() == 'sunos'
+  defines += [
+   'OPENSSLDIR="/etc/ssl"',
+   'ENGINESDIR="/dev/null"',
+   '__EXTENSIONS__'
+ ]
+else
+  # linux and others
+  defines += [
+    'OPENSSLDIR="/etc/ssl"',
```

Comment:
This should be `'OPENSSLDIR="@0@/ssl"'.format(get_option('sysconfdir'))`. Also above for sunos (so probably best to use a common `openssldir` variable.

`sysconfdir` is a magic option that is `$PREFIX/etc` in general and is `/etc` when `$PREFIX` is `/usr`.

Reply:
Is this Linux-specific? I didn't find anything mentioning its behavior on Windows/macOS.

Also I'm concerned that it might not work properly unless configured with `/usr` when OpenSSL used as a dependency by another library and expects system-wide certificates to just work. Maybe worth having some option for this instead?